Lib/javascript/v8/javascriptinit.swg

```diff
@@ -120,7 +173,14 @@ void SWIGV8_INIT (SWIGV8_OBJECT exports_obj, SWIGV8_VALUE /*module*/, v8::Local<
 #if (NODE_MODULE_VERSION < 64)
 NODE_MODULE($jsname, $jsname_initialize)
 #else
-NODE_MODULE_CONTEXT_AWARE($jsname, $jsname_initialize)
+#ifndef NODE_GYP_MODULE_NAME
+# define NODE_GYP_MODULE_NAME $jsname
+#endif
+#if defined(__GNUC__) && __GNUC__>=8
+// NODE_MODULE_INIT is not self-consistent in all supported Node.js versions :-(
+# pragma GCC diagnostic ignored "-Wcast-function-type"
```

Comment:
It would be cleaner to `# pragma GCC diagnostic push` before this, then after the place we want to suppress the warning to repeat the `#if` conditonal with `# pragma GCC diagnostic pop` inside.

Is this needed for clang too?  All non-antique clang versions currently pretend to be GCC 4.2.1 by default.

It would also be good to note in a comment the node version at which this became consistent so that when SWIG raises its minimum supported node version we know whether this workaround is still needed.  Otherwise it'll end up lingering for years after it could have been removed.

Reply:
> It would be cleaner to `# pragma GCC diagnostic push` before this,

Rationale behind not making it more elaborate than necessary was that it's **known** to be the very end of the compilation unit.

> Is this needed for clang too?

Not that I could tell.

Reply:
> It would also be good to note in a comment the node version at which this became consistent so that when SWIG

Commentary says "**all** supported versions." And as far as I can see the problem persists with latest version.

Reply:
> Rationale behind not making it more elaborate than necessary was that it's known to be the very end of the compilation unit.

The `post-init` goes after this (it's just empty by default):

```
$ git diff
diff --git a/Examples/javascript/simple/example.i b/Examples/javascript/simple/example.i
index 24093b9b..35c1ab8d 100644
--- a/Examples/javascript/simple/example.i
+++ b/Examples/javascript/simple/example.i
@@ -1,6 +1,9 @@
 /* File : example.i */
 %module example
 
+%insert("post-init") {
+// THIS GOES LAST
+}
 %inline %{
 extern int    gcd(int x, int y);
 extern double Foo;
$ make -s -C Examples/javascript/simple
GCD of x and y is=21
Global variable Foo=3
Variable Foo changed to=3.1415926
$ tail Examples/javascript/simple/example_wrap.cxx 
#if defined(BUILDING_NODE_EXTENSION)
#if (NODE_MODULE_VERSION < 64)
NODE_MODULE(example, example_initialize)
#else
NODE_MODULE_CONTEXT_AWARE(example, example_initialize)
#endif
#endif

// THIS GOES LAST

```

> Commentary says "**all** supported versions." And as far as I can see the problem persists with latest version.

Oh, I see - then I'd suggest rewording as "NODE_MODULE_INIT is not self-consistent in any supported Node.js version" then - I interpreted it as "it's not the case that: NODE_MODULE_INIT is self-consistent in all supported Node.js versions" (but it is in some)...

Reply:
Added push/pop.